### PR TITLE
Refactor Scalar DB proto

### PR DESCRIPTION
**Pull request by brfrn169**
This PR basically renames the fields of protocol buffers messages. Since it just does renaming, it should not violate backward compatibility. I added several inline comments for the details. Please take a look!

### CODE REVIEW

File: rpc/src/main/proto/scalardb.proto [hunk 1]
```diff
@@ -4,7 +4,7 @@ option java_multiple_files = true;
 option java_package = "com.scalar.db.rpc";
 option java_outer_classname = "ScalarDbProto";
 
-package rpc;
+package scalardb.rpc;
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:7]: Changed the package name.

File: rpc/src/main/proto/scalardb.proto [hunk 2]
```diff
@@ -46,7 +46,7 @@ message Column {
 
 message Key {
   repeated Value value = 1 [deprecated = true];
-  repeated Column column = 2;
+  repeated Column columns = 2;
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:49]: Made the names `repeated` of fields plural.

File: rpc/src/main/proto/scalardb.proto [hunk 3]
```diff
@@ -211,11 +211,11 @@ enum DataType {
 }
 
 message TableMetadata {
-  map<string, DataType> column = 1;
-  repeated string partition_key_name = 2;
-  repeated string clustering_key_name = 3;
-  map<string, Order> clustering_order = 4;
-  repeated string secondary_index_name = 5;
+  map<string, DataType> columns = 1;
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:214]: Also made the names of `map` fields plural.

File: rpc/src/main/proto/scalardb.proto [hunk 4]
```diff
@@ -357,13 +356,13 @@ message TransactionResponse {
...
   message ScanResponse {
-    repeated Result result = 1;
+    repeated Result results = 1;
   }
 
   message Error {
     enum ErrorCode {
       INVALID_ARGUMENT = 0;
-      CONFLICT = 1;
+      TRANSACTION_CONFLICT = 1;
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:365]: Renamed `CONFLICT` to `TRANSACTION_CONFLICT`.

File: rpc/src/main/proto/scalardb.proto [hunk 5]
```diff
@@ -468,14 +467,14 @@ message TwoPhaseCommitTransactionResponse {
...
+    repeated Result results = 1;
   }
 
   message Error {
     enum ErrorCode {
       INVALID_ARGUMENT = 0;
-      CONFLICT = 1;
-      UNKNOWN_TRANSACTION = 2;
+      TRANSACTION_CONFLICT = 1;
+      UNKNOWN_TRANSACTION_STATUS = 2;
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:477]: Also renamed `UNKNOWN_TRANSACTION` to `UNKNOWN_TRANSACTION_STATUS`.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

I feel adding only one element with `addXXXs` method is pretty unusual, but seems like it's the way of the protobuf ...

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Comment by brfrn169** [comment]
> I feel adding only one element with `addXXXs` method is pretty unusual, but seems like it's the way of the protobuf ...

Yeah, I thought so too. [The protocol-buffers style guide](https://developers.google.com/protocol-buffers/docs/style#repeated_fields) says we should do that...

**Pull request merged**